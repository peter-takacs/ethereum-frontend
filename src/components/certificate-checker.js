import React from 'react';

const CertificateChecker = ({hasCertificate, onClick, candidate, certificate, onChange}) => {
    let addressInput;
    let certificateInput;

    let handleChange = () => {
        onChange(addressInput.value, certificateInput.value);
    }

    return (
        <div>
            <header>
            Certificate checker 
            </header>
            <div>
            <input type="text" id="address" placeholder="Candidate address" value={candidate} ref={node => addressInput = node} onChange={handleChange}/>
            <input type="text" id="certificate" placeholder="Certificate to check" value={certificate} ref={node => certificateInput = node} onChange={handleChange}/>
            <button onClick={() => onClick(candidate, certificate)}>Submit</button>
            </div>
        </div>
    );
}

export default CertificateChecker;