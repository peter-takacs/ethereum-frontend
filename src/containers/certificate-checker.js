import { connect } from 'react-redux';
import CertificateChecker from '../components/certificate-checker';
import { getStatus, changeQuery } from '../actions/certificate-holder-actions';

const mapDispatchToProps = dispatch => ({
    onClick: (candidate, certificate) => dispatch(getStatus(candidate, certificate)),
    onChange: (candidate, certificate) => dispatch(changeQuery(candidate, certificate))
})

const mapStateToProps = (state) => {
    return {
        candidate: state.certificateChecker.candidate,
        hasCertificate: state.certificateChecker.hasCertificate,
        certificate: state.certificateChecker.certificate
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(CertificateChecker);