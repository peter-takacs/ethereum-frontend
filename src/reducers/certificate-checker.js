
import {
    RECEIVE_STATUS,
    REQUEST_STATUS
} from '../actions';
import { CHANGE_QUERY } from '../actions/certificate-holder-actions';

const certificateChecker = (state = {candidate: '', hasCertificate: false, certificate: ''}, action) => {
    switch (action.type) {
        case RECEIVE_STATUS:
            return {
                ...state,
                hasCertificate: action.hasCertificate 
            }
        case REQUEST_STATUS:
            return {
                ...state,
                candidate: action.candidate,
                certificate: action.certificate
            }
        case CHANGE_QUERY:
            return {
                ...state,
                candidate: action.candidate,
                certificate: action.certificate
            }
        default:
            return state;
    }
}

export default certificateChecker;