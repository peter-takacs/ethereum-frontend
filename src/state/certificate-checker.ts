import { Address } from "../types/ethereum-address";

export enum CertificateCheckState {
    Idle,
    Checking,
    Confirmed,
    Rejected
}
export interface State {
    candidate: Address | null,
    certificate: string,
    certificateStatus: CertificateCheckState
}