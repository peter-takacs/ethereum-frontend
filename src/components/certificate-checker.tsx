import * as React from 'react';
import { State, CertificateCheckState } from '../state/certificate-checker';
import { Input, FormControl, Button, TextField, Grid } from '@material-ui/core';
import AddressEditor from './address-editor';
import { Address } from '../types/ethereum-address';

export interface CertificateCheckerDispatch {
    onClick: (candidate: Address, certificate: string) => void;
    onCandidateChange: (candidate: Address) => void;
    onCertificateChange: (certificate: string) => void;
}

export type CertificateCheckerProps = State & CertificateCheckerDispatch;

export class CertificateChecker extends React.Component<CertificateCheckerProps> {
    private readonly onClick: (candidate: Address, certificate: string) => void;
    private readonly candidate: Address | null;
    private readonly certificate: string;
    private readonly onCandidateChange: (candidate: Address) => void;
    private readonly onCertificateChange: (certificate: string) => void;
    private readonly resultText: string;


    private candidateChange(address: Address) {
        this.onCandidateChange(address);
    }
    private certificateChange(event: any) {
        this.onCertificateChange(event.target.value);
    }

    private get isValid() {
        if (this.candidate == null || this.certificate == null) {
            return;
        }
    }

    private onSubmitClicked() {
        if (this.isValid) {
            this.onClick(this.candidate!, this.certificate);
        }
    }

    constructor(props: CertificateCheckerProps) {
        super(props);
        this.onClick = props.onClick;
        this.candidate = props.candidate;
        this.certificate = props.certificate;
        this.onCandidateChange = props.onCandidateChange;
        this.onCertificateChange = props.onCertificateChange;
        switch (props.certificateStatus) {
            case CertificateCheckState.Checking:
                this.resultText = 'Loading...';
                break;
            case CertificateCheckState.Confirmed:
                this.resultText = 'The given address has the certificate';
                break;
            case CertificateCheckState.Rejected:
                this.resultText = 'The given addres does not have the certificate';
                break;
            default:
                this.resultText = '';
                break;
        }

    }

    render() {
        return (
            <Grid container spacing={24}>
                <Grid item xs>
                    <FormControl>
                        <AddressEditor
                            placeholder="Candidate address"
                            address={this.candidate}
                            onChange={this.candidateChange}
                        />
                        <TextField 
                            id="certificate" 
                            placeholder="Certificate to check" 
                            value={this.certificate} 
                            onChange={this.certificateChange} 
                        />
                        <Button onClick={() => this.onSubmitClicked()}>Submit</Button>
                    </FormControl>
                </Grid>
                <Grid item xs>
                    {this.resultText}
                </Grid>
            </Grid>
        );
    }
}

export default CertificateChecker;