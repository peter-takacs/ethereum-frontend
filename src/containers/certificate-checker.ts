import { connect } from 'react-redux';
import CertificateChecker, { CertificateCheckerDispatch } from '../components/certificate-checker';
import { getStatus, changeQuery, Actions } from '../actions/certificate-holder-actions';
import { State as RootState } from '../state/root';
import { State as CertificateCheckerState} from '../state/certificate-checker';
import { ThunkDispatch } from 'redux-thunk';
import { Address } from '../types/ethereum-address';

const mapDispatchToProps = (dispatch: ThunkDispatch<CertificateCheckerState, undefined, Actions>) => ({
    onClick: (candidate: Address, certificate: string) => dispatch(getStatus({candidate, certificate})),
    onCandidateChange: (candidate: Address) => dispatch(changeQuery({candidate})),
    onCertificateChange: (certificate: string) => dispatch(changeQuery({certificate})),
})

const mapStateToProps = (state: RootState): CertificateCheckerState => {
    return {
        candidate: state.certificateChecker.candidate,
        certificateStatus: state.certificateChecker.certificateStatus,
        certificate: state.certificateChecker.certificate
    }
}
export default connect<CertificateCheckerState, CertificateCheckerDispatch, {}, RootState>(
    mapStateToProps, 
    mapDispatchToProps
)(CertificateChecker);