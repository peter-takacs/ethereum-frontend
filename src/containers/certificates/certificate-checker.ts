import { connect } from 'react-redux';
import CertificateChecker, { CertificateCheckerDispatch } from '../../components/certificates/certificate-checker';
import { getStatus, Actions } from '../../actions/certificate-holder-actions';
import { State as RootState } from '../../state/root';
import { CertificateCheckerState } from '../../state/certificate-checker';
import { ThunkDispatch } from 'redux-thunk';
import { Address } from '../../types/ethereum-address';

const mapDispatchToProps = (dispatch: ThunkDispatch<CertificateCheckerState, undefined, Actions>) => ({
    onClick: (candidate: Address, certificate: string) => dispatch(getStatus({candidate, certificate})),
})

const mapStateToProps = (state: RootState): CertificateCheckerState => {
    return {
        certificateStatus: state.certificateChecker.certificateStatus,
    }
}
export default connect<CertificateCheckerState, CertificateCheckerDispatch, {}, RootState>(
    mapStateToProps, 
    mapDispatchToProps
)(CertificateChecker);