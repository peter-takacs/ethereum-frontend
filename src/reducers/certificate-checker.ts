
import { Actions, RECEIVE_STATUS, CHANGE_STATUS_QUERY, REQUEST_STATUS } from '../actions/certificate-holder-actions';
import { State, CertificateCheckState } from '../state/certificate-checker';
const defaultState = {candidate: null, certificateStatus: CertificateCheckState.Idle, certificate: ''};
const certificateChecker = (state: State = defaultState, action: Actions): State => {
    switch (action.type) {
        case REQUEST_STATUS:
            return {
                ...state,
                certificateStatus: CertificateCheckState.Checking
            }
        case RECEIVE_STATUS:
            return {
                ...state,
                certificateStatus: action.hasCertificate
                    ? CertificateCheckState.Confirmed
                    : CertificateCheckState.Rejected
            }
       case CHANGE_STATUS_QUERY:
            return {
                ...state,
                candidate: action.query.candidate == null 
                    ? state.candidate
                    : action.query.candidate,
                certificate: action.query.certificate == null
                    ? state.certificate
                    : action.query.certificate
            }
        default:
            return state;
    }
}

export default certificateChecker;