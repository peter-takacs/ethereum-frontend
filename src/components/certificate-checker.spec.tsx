

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CertificateChecker from './certificate-checker';
import { CertificateCheckState } from '../state/certificate-checker';
import { dummyAddress } from '../utils/test/address';

test('Renders correctly in every state', () => {
    [
        CertificateCheckState.Idle,
        CertificateCheckState.Checking,
        CertificateCheckState.Confirmed,
        CertificateCheckState.Rejected
    ].forEach(state => {
        const component = renderer.create(
            <CertificateChecker
                onClick={() => { }}
                certificateStatus={state} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})