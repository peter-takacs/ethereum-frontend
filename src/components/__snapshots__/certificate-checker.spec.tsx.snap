// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly in every state 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xABCD"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to check"
            required={false}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    
  </div>
</div>
`;

exports[`Renders correctly in every state 2`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xABCD"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to check"
            required={false}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Loading...
  </div>
</div>
`;

exports[`Renders correctly in every state 3`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xABCD"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to check"
            required={false}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    The given address has the certificate
  </div>
</div>
`;

exports[`Renders correctly in every state 4`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xABCD"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to check"
            required={false}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
        <span
          className="MuiTouchRipple-root-144"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    The given addres does not have the certificate
  </div>
</div>
`;
